### ISSUE TITLE
Query batching not working when using fragments

### ISSUE THREAD

**Issue by jordan-burnett** [apollostack/apollo-client#501]
```
query {
    neighbourhoods {
        id
        ...neighbourhoodMap
    }
 }
```

Gives me the following error when batched with another query:

![image](https://cloud.githubusercontent.com/assets/2143374/17373661/1e511bec-59a1-11e6-9386-a74ea89be392.png)

Works fine when batching is disabled or the fragments are removed.

**Comment by Poincare** [comment]
Could you describe how you're calling `query` or `watchQuery`?

**Comment by jordan-burnett** [comment]
Apologies, I'm using `react-apollo` and `mapQueriesToProps`

**Comment by Poincare** [comment]
What does the fragment `neighbourhoodMap` look like? Are you referencing it through the return value of `createFragment` or is it embedded within the query document?

**Comment by jordan-burnett** [comment]
So neighbourhoodMap looks like:

```
const neighbourhoodMapFragment = createFragment(gql`
  fragment neighbourhoodMap on Neighbourhood {
    id
    name
    region {
      encodedShape
      centerPoint
    }
  }
`);
```

The full query looks like:
```
function mapQueriesToProps({ ownProps, state }) {
  return {
    neighbourhoods: {
      query: gql`
        query {
          neighbourhoods {
            id
            ...neighbourhoodMap
          }
        }
      `,
      fragments: neighbourhoodMapFragment
    },
  };
};
```

**Comment by Poincare** [comment]
Ah, this does look like a bug. I will look into it and see if I can reproduce it with a test.

**Comment by Poincare** [comment]
This turned out to be a pretty involved bug to fix but I think #509 should be the fix for it.

**Comment by Poincare** [comment]
Fixed by #509.
